### PR TITLE
649 upgrade node version to 10.15.3 in Dockefile and circleCI config

### PR DESCRIPTION
**Pull request by lewisdaly**
Changes for mojaloop/project#649

Upgrades the node version to 10.15.1

Also included in PR:
- fix onboarding documentation 'mangodb' typo
- change mentioned node version in onboarding docs
- update the package-lock.json: it seems that this hasn't been updated in a while

### CODE REVIEW

**Comment by lewisdaly** [comment]
Still need to update the dependencies.

File: test/unit/lib/database.test.js [hunk 1]
```diff
@@ -86,12 +86,7 @@ test('Mongo Database tests', async dbTest => {
       let result = await Database()
       assert.deepEquals(result.$initialConnection, expectedResult.$initialConnection)
       assert.pass('Db connection successful')
-      process.nextTick(() => {
```
Comment (lewisdaly) [inline test/unit/lib/database.test.js]: This was closing the database connection too early, and the last 3 tests were failing. I'm not sure what purpose it has as the node process will exit with 0 should the tests pass (someone please correct me if I'm wrong).

I tested with the older version of node as well, and it also failed. Either there's something wrong with my environment, or this test has been broken for a little while.

Comment (vgenev) [reply test/unit/lib/database.test.js]: tests are not working properly on that one. I have some more tests in my local fork, but still not at the coverage we are aiming at. do not mind the tests atm.

Comment (lewisdaly) [reply test/unit/lib/database.test.js]: Ok no problem. I'll revert my changes.

**Comment by lewisdaly** [comment]
> Please include PR reference in story #649 and move to REVIEW/QA pipeline.

Yeah I tried earlier, but I don't have permission to do either of those things.

**Comment by lewisdaly** [comment]
@elnyry, can you please merge this?

**Pull request merged**